tests/client/components/carousel.spec.js

```diff
@@ -5,126 +5,434 @@ import { mount } from '@vue/test-utils';
-    const wrapper = mount(Carousel);
+  describe('Adjustable height easing', () => {
+    it('should not set adjustable height transition easing', done => {
+      const wrapper = mount(Carousel, {
+        propsData: {
+          adjustableHeight: false,
+          adjustableHeightEasing: 'linear',
+          speed: 0
+        }
+      });
 
-    const carouselInstance = wrapper.vm;
-    carouselInstance.$destroy();
-    expect(carouselInstance._isDestroyed).toBeTruthy();
+      expect(wrapper.vm.transitionStyle).toBe('0s ease transform');
 
-    expect(wrapper).toMatchSnapshot();
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper).toMatchSnapshot();
+        done();
+      });
+    });
+
+    it('should set adjustable height transition easing', done => {
+      const wrapper = mount(Carousel, {
+        propsData: {
+          adjustableHeight: true,
+          adjustableHeightEasing: 'linear',
+          speed: 0
+        }
+      });
+
+      expect(wrapper.vm.transitionStyle).toBe('0s ease transform, height 0s linear');
+
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper).toMatchSnapshot();
+        done();
+      });
+    });
   });
 
-  it('should be unable to advance backward by default', () => {
-    const wrapper = mount(Carousel);
+  describe('Autoplay functionality', () => {
+    it.skip('should disable autoplay by default', () => {});
+
+    it.skip('should set the autoplay direction to forward by default', () => {});
+
+    it.skip('should set the autoplay hover pause to true by default', () => {});
+
+    it.skip('should set the autoplay timeout to 2000 by default', () => {});
+
+    it('should begin autoplaying when option specified', done => {
+      const wrapper = mount(Carousel, {
+        propsData: {
+          perPage: 1,
+          autoplay: true,
+          autoplayHoverPause: true
+        },
+        slots: {
+          default: [Slide, Slide]
+        }
+      });
+
+      expect(wrapper.vm.autoplayInterval).toBeDefined();
+      wrapper.vm.pauseAutoplay();
+      expect(wrapper.vm.autoplayInterval).toBe(undefined);
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper).toMatchSnapshot();
+        done();
+      });
+    });
+
+    // start on first slide and allow the autoplay to advance to the second slide
+    it.skip('should display the second slide after waiting the timeout duration with direction forward', () => {});
+
+    // start on the second slide and allow autoplay to advanced to the first slide
+    it.skip('should display the first slide after waiting the timeout duration with direction backward', () => {});
+
+    it.skip('should not allow the autoplay feature to advance on hover', () => {});
+
+    it.skip('should allow the autoplay feature to advance on hover when hover pause false', () => {});
 
-    expect(wrapper.vm.canAdvanceBackward).toBeFalsy();
+    it.skip('should show the first slide twice when looping is enabled with autoplay', () => {});
```

Comment:
What does `show slide twice` mean? Once at the start and once at the end of the array?

Reply:
Yeah something like that, moved to a todo for now anyway